Enable "tagliatelle" linter in golangci-lint

Enable "tagliatelle" linter in golangci-lint. Fix any warnings enabling it causes.

/area ci
/help
/triage accepted

@tuminoid  regarding this linter, i did quick look:
```
cmd/make-virt-host/main.go:118:14: json(camel): got 'Port' want 'port' (tagliatelle)
        Port int    `json:"Port"`
                    ^
cmd/make-virt-host/main.go:119:14: json(camel): got 'Domain name' want 'domainName' (tagliatelle)
        Name string `json:"Domain name"`
                    ^
```

and it seems project already rely on this format, if i change it we need to update comments, may be docs?

https://github.com/metal3-io/baremetal-operator/blob/e609781c5d36b18f75ddcef8d871dd3117275c36/cmd/make-virt-host/main.go#L105-L114

so let me know if it's really make sense to change and enable linter.

--- a/.golangci.yaml
+++ b/.golangci.yaml
@@ -64,7 +64,7 @@ linters:
   - rowserrcheck
   - staticcheck
   - stylecheck
-  #- tagliatelle
+  - tagliatelle
   - testifylint
   - thelper
   - tparallel
@@ -124,6 +124,10 @@ linters-settings:
     - unnecessaryDefer
     - whyNoLint
     - wrapperFunc
+  tagliatelle:
+    case:
+      rules:
+        json: goCamel
 issues:
   exclude-dirs:
   - mock*
@@ -199,6 +203,9 @@ issues:
     - stylecheck
     text: "ST1016: methods on the same type should have the same receiver name"
     path: .*(api|types)\/.*\/conversion.*\.go$
+  - linters:
+    - tagliatelle
+    text: "RAID|MAC|BMO"
   include:
   - EXC0002 # include "missing comments" issues from golangci-lint
   max-issues-per-linter: 0

--- a/apis/metal3.io/v1alpha1/baremetalhost_types.go
+++ b/apis/metal3.io/v1alpha1/baremetalhost_types.go
@@ -679,6 +679,7 @@ type NIC struct {
 	VLANs []VLAN `json:"vlans,omitempty"`
 
 	// The untagged VLAN ID
+	//nolint:tagliatelle
 	VLANID VLANID `json:"vlanId,omitempty"`
 
 	// Whether the NIC is PXE Bootable
@@ -866,6 +867,7 @@ type ProvisionStatus struct {
 
 	// The hosts's ID from the underlying provisioning tool (e.g. the
 	// Ironic node UUID).
+	//nolint:tagliatelle
 	ID string `json:"ID"`
 
 	// Image holds the details of the last image successfully

--- a/apis/metal3.io/v1alpha1/firmwareschema_types.go
+++ b/apis/metal3.io/v1alpha1/firmwareschema_types.go
@@ -30,24 +30,31 @@ type SettingSchema struct {
 
 	// The type of setting.
 	// +kubebuilder:validation:Enum=Enumeration;String;Integer;Boolean;Password
+	//nolint:tagliatelle
 	AttributeType string `json:"attribute_type,omitempty"`
 
+	//nolint:tagliatelle
 	// The allowable value for an Enumeration type setting.
 	AllowableValues []string `json:"allowable_values,omitempty"`
 
 	// The lowest value for an Integer type setting.
+	//nolint:tagliatelle
 	LowerBound *int `json:"lower_bound,omitempty"`
 
 	// The highest value for an Integer type setting.
+	//nolint:tagliatelle
 	UpperBound *int `json:"upper_bound,omitempty"`
 
 	// Minimum length for a String type setting.
+	//nolint:tagliatelle
 	MinLength *int `json:"min_length,omitempty"`
 
 	// Maximum length for a String type setting.
+	//nolint:tagliatelle
 	MaxLength *int `json:"max_length,omitempty"`
 
 	// Whether or not this setting is read only.
+	//nolint:tagliatelle
 	ReadOnly *bool `json:"read_only,omitempty"`
 
 	// Whether or not this setting's value is unique to this node, e.g.

--- a/apis/metal3.io/v1alpha1/preprovisioningimage_types.go
+++ b/apis/metal3.io/v1alpha1/preprovisioningimage_types.go
@@ -63,11 +63,13 @@ const (
 // PreprovisioningImageStatus defines the observed state of PreprovisioningImage.
 type PreprovisioningImageStatus struct {
 	// imageUrl is the URL from which the built image can be downloaded.
+	//nolint:tagliatelle
 	ImageUrl string `json:"imageUrl,omitempty"` //nolint:revive,stylecheck
 
 	// kernelUrl is the URL from which the kernel of the image can be downloaded.
 	// Only makes sense for initrd images.
 	// +optional
+	//nolint:tagliatelle
 	KernelUrl string `json:"kernelUrl,omitempty"` //nolint:revive,stylecheck
 
 	// extraKernelParams is a string with extra parameters to pass to the

--- a/cmd/make-virt-host/main.go
+++ b/cmd/make-virt-host/main.go
@@ -115,7 +115,9 @@ vbmc list -f json -c 'Domain name' -c Port
 // VBMC holds the parameters for describing a virtual machine
 // controller.
 type VBMC struct {
-	Port int    `json:"Port"`
+	//nolint:tagliatelle
+	Port int `json:"Port"`
+	//nolint:tagliatelle
 	Name string `json:"Domain name"`
 }
 

--- a/pkg/provisioner/ironic/ironic.go
+++ b/pkg/provisioner/ironic/ironic.go
@@ -105,8 +105,9 @@ type ironicProvisioner struct {
 // FIXME(hroyrh) : move this to gophercloud when implementing
 // VirtualMediaGetResult Extract function.
 type VirtualMedia struct {
-	Image      string   `json:"image"`
-	Inserted   bool     `json:"inserted"`
+	Image    string `json:"image"`
+	Inserted bool   `json:"inserted"`
+	//nolint:tagliatelle
 	MediaTypes []string `json:"media_types"`
 }
 

